### ISSUE TITLE
Add cursor components mixin to Game

### ISSUE THREAD

**Issue by garysm** [flame-engine/flame#1818]
# Problem to solve

It would be convenient to propagate PointerHoverEvents to child Components, not just handle them from the base game.

# Proposal

Make a HasCursorHandlerComponents mixin on FlameGame, could be very similar to HasKeyboardHandlerComponents.  See example below:
```dart
mixin HasCursorHandlerComponents on FlameGame implements MouseMovementDetector {
  @override
  @mustCallSuper
  void onMouseMove(PointerHoverInfo info) {
    propagateToChildren<CursorHandler>(
      (CursorHandler child) => child.onMouseMove(info),
    );
  }
}
```
# More information

Interested in a PR, basic implementation already tested.

**Comment by spydon** [comment]
Hi, we already have this. :)
The mixin for the components is called `Hoverable` and for the game it is `HasHoverableComponents`.

**Comment by spydon** [comment]
Ah, or you mean that it should propagate all mouse movement to all children all the time even when the pointer is not within the bounds of the component?

**Comment by garysm** [comment]
Correct, the latter.  It would be useful to handle the mouse movement from each component to do interesting things like updating the angle of components according to where the cursor is hovering.
